client/my-sites/marketplace/pages/marketplace-plugin-install/index.tsx

```diff
@@ -256,17 +259,26 @@ const MarketplacePluginInstall = ( {
 				/>
 			);
 		}
-		if ( noDirectAccessError ) {
+		if ( noDirectAccessError && ! directInstallationAllowed ) {
 			return (
 				<EmptyContent
-					illustration="/calypso/images/illustrations/error.svg"
-					title={ null }
-					line={ translate(
-						'This URL should not be accessed directly. Please click the Install button on the plugin page.'
-					) }
-					action={ translate( 'Go to the plugin page' ) }
-					actionURL={ `/plugins/${ productSlug }/${ selectedSite?.slug }` }
-				/>
+					className="marketplace-plugin-install__direct-install-container"
+					illustration={ wporgPlugin?.icon || '/calypso/images/illustrations/error.svg' }
+					illustrationWidth={ wporgPlugin?.icon && 128 }
+					title={ wporgPlugin?.name || productSlug }
+					line={ translate( 'Do you want to install the plugin %(plugin)s?', {
+						args: { plugin: wporgPlugin?.name || productSlug },
+					} ) }
+				>
+					<div className="marketplace-plugin-install__direct-install-actions">
+						<Button href={ `/plugins/${ productSlug }/${ selectedSite?.slug }` }>
+							{ translate( 'Go to the plugin page' ) }
+						</Button>
+						<Button primary onClick={ () => setDirectInstallationAllowed( true ) }>
+							{ translate( 'Install plugin' ) }
+						</Button>
+					</div>
```

Comment:
Have you tried using the `EmptyContent` props like here
https://github.com/Automattic/wp-calypso/blob/ba2918dad5f3fb44745ad62a6b3a0f88eb8b6a89/client/my-sites/marketplace/pages/marketplace-plugin-install/index.tsx#L280-L283

instead of adding markup? They support a `Callback` too.

Reply:
> Probably, uploading a plugin doesn't even need to be guarded, right @gcsecsey?

Yes, I don't think it's necessary to guard that flow.

Reply:
> Have you tried using the EmptyContent props

Yes, I have tried, but it is not possible to make the primary button goes on the right of the secondary button with EmptyContent, that is why I added manually.